raw/src/requests/restrict_chat_member.rs

```diff
@@ -0,0 +1,63 @@
+use types::*;
+use requests::*;
+
+/// Use this method to kick a user from a group or a supergroup.
+/// In the case of supergroups, the user will not be able to return to the group on
+/// their own using invite links, etc., unless unbanned first.
+/// The bot must be an administrator in the group for this to work.
+#[derive(Debug, Clone, PartialEq, PartialOrd, Serialize)]
+#[must_use = "requests do nothing unless sent"]
+pub struct RestrictChatMember {
+    chat_id: ChatRef,
+    user_id: UserId,
+    until_date: Option<i32>,
+    can_send_messages: Option<bool>,
+    can_send_media_messages: Option<bool>,
+    can_send_other_messages: Option<bool>,
+    can_add_web_page_previews: Option<bool>,
+}
+
+impl Request for RestrictChatMember {
+    type Type = JsonRequestType<Self>;
+    type Response = JsonTrueToUnitResponse;
+
+    fn serialize(&self) -> Result<HttpRequest, Error> {
+        Self::Type::serialize(RequestUrl::method("restrictChatMember"), self)
+    }
+}
+
+impl RestrictChatMember {
+    pub fn new<C, U>(chat: C, user: U, until_date: Option<i32>, can_send_messages: Option<bool>, can_send_media_messages: Option<bool>, can_send_other_messages: Option<bool>, can_add_web_page_previews: Option<bool>) -> Self where C: ToChatRef, U: ToUserId {
```

Comment:
I think it's too many arguments for one method. How about `pub fn new<C, U>(chat: C, user: U)` and something like `pub fn can_send_messages(&mut self, value: bool) -> &mut Self`?

Reply:
Fixed!